### PR TITLE
feat: add quantity display in recipe lists on ingredient pages

### LINKED ISSUE #47: Feat: Add quantity of ingredient within lists
Within the list of recipes using a specific ingredient, I want to see the quantity each recipe calls for.

For examples on https://sboudrias.github.io/cocktails/ingredient/puree/guava-puree

We'd like to see how much Guava Purée each recipe needs.

### PR DESCRIPTION
**Pull request by SBoudrias**
## Summary
- Add `tertiary` prop to `LinkListItem` for right-aligned content next to primary text
- Display ingredient quantities in recipe lists on ingredient pages
- Display ingredient quantities in recipe lists on category pages
- Show attribution for recipes with duplicate names
- Fix Quantity component to render as span for valid HTML nesting
- Add test suite for category pages

Closes #47

## Test plan
- [x] Verify ingredient pages show quantities for each recipe
- [x] Verify category pages show quantities for each recipe
- [x] Verify attribution shown for duplicate recipe names
- [x] Run `yarn vitest --run` - all 178 tests pass
- [x] Run `yarn lint` - no errors

### CODE REVIEW

File: src/app/ingredient/[type]/[slug]/page.tsx [hunk 1]
```diff
@@ -170,6 +171,12 @@ export default async function IngredientPage({ params }: { params: Promise<Param
         <RecipeList
           recipes={relatedRecipes}
           header={`Recipes calling for ${ingredient.name}`}
+          getSecondaryContent={(recipe) => {
```
Comment (SBoudrias) [inline src/app/ingredient/[type]/[slug]/page.tsx]: Let's add page level tests to cover that feature.

File: src/app/ingredient/[type]/[slug]/page.tsx [hunk 2]
```diff
@@ -170,6 +171,12 @@ export default async function IngredientPage({ params }: { params: Promise<Param
...
+            const ing = recipe.ingredients.find((i) => i.slug === ingredient.slug);
+            return ing ? (
+              <Quantity quantity={ing.quantity} preferredUnit="oz" />
+            ) : undefined;
```
Comment (SBoudrias) [inline src/app/ingredient/[type]/[slug]/page.tsx]: Let's combine both the quantity and `getRecipeAttribution` is name isn't unique.

Like `3oz | Smuggler's Cove`

File: src/app/ingredient/[type]/[slug]/page.tsx [hunk 3]
```diff
@@ -170,6 +171,22 @@ export default async function IngredientPage({ params }: { params: Promise<Param
         <RecipeList
           recipes={relatedRecipes}
           header={`Recipes calling for ${ingredient.name}`}
+          getSecondaryContent={(recipe, recipes) => {
+            const ing = recipe.ingredients.find((i) => i.slug === ingredient.slug);
+            if (!ing) return undefined;
+
+            const nameCount = recipes.filter(
+              (r) => r.name.toLowerCase() === recipe.name.toLowerCase(),
+            ).length;
```
Comment (SBoudrias) [inline src/app/ingredient/[type]/[slug]/page.tsx]: This should reuse the logic of `nameIsUnique` - let's make a reusable utility we can reuse here.

File: src/app/ingredient/[type]/[slug]/page.test.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  getIngredient,
+  getRecipesForIngredient,
+  getSubstitutesForIngredient,
+} from '@/modules/ingredients';
+
+vi.mock('@/modules/ingredients', () => ({
+  getIngredient: vi.fn(),
+  getRecipesForIngredient: vi.fn(),
+  getSubstitutesForIngredient: vi.fn(),
+}));
```
Comment (SBoudrias) [inline src/app/ingredient/[type]/[slug]/page.test.tsx]: Let's not mock our data getter. Instead use an ingredient with enough overlap to test real edge cases.

I recommend `pineapple juice`.

File: src/app/ingredient/[type]/[slug]/page.test.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import { describe, it, expect } from 'vitest';
+import { screen } from '@testing-library/react';
+import IngredientPage from './page';
+import { setupApp } from '@/testing';
+
+describe('IngredientPage', () => {
```
Comment (SBoudrias) [inline src/app/ingredient/[type]/[slug]/page.test.tsx]: Those tests are slow because they need to load data. In our test setup script, let's trigger our data loader so our data is pre-memoized when come time to run tests.

File: src/app/ingredient/[type]/[slug]/page.tsx [hunk 6]
```diff
@@ -170,6 +172,21 @@ export default async function IngredientPage({ params }: { params: Promise<Param
         <RecipeList
           recipes={relatedRecipes}
           header={`Recipes calling for ${ingredient.name}`}
+          getSecondaryContent={(recipe) => {
+            const ing = recipe.ingredients.find((i) => i.slug === ingredient.slug);
+            if (!ing) return undefined;
```
Comment (SBoudrias) [inline src/app/ingredient/[type]/[slug]/page.tsx]: We should pursue with the attribution logic even if we don't find the ingredient.

File: src/app/ingredient/[type]/[slug]/page.tsx [hunk 7]
```diff
@@ -170,6 +172,21 @@ export default async function IngredientPage({ params }: { params: Promise<Param
...
+
+            const attribution = nameIsUnique(recipe, relatedRecipes)
+              ? undefined
+              : getRecipeAttribution(recipe);
+
+            return (
+              <>
+                <Quantity quantity={ing.quantity} preferredUnit="oz" />
+                {attribution && ` | ${attribution}`}
```
Comment (SBoudrias) [inline src/app/ingredient/[type]/[slug]/page.tsx]: This gets displayed on 2 lines instead of side-by-side.
